packages/test/test-end-to-end-tests/src/test/waitForSummary.spec.ts

```diff
@@ -0,0 +1,220 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { bufferToString, stringToBuffer } from "@fluid-internal/client-utils";
+import { describeCompat } from "@fluid-private/test-version-utils";
+import {
+	LoaderHeader,
+	type IContainer,
+	type IHostLoader,
+} from "@fluidframework/container-definitions/internal";
+import type { IContainerExperimental } from "@fluidframework/container-loader/internal";
+import type {
+	ConfigTypes,
+	IConfigProviderBase,
+	IRequestHeader,
+} from "@fluidframework/core-interfaces";
+import type { ISharedMap } from "@fluidframework/map/internal";
+import {
+	type ITestObjectProvider,
+	type ITestContainerConfig,
+	createSummarizer,
+	summarizeNow,
+	type ChannelFactoryRegistry,
+	createAndAttachContainer,
+	DataObjectFactoryType,
+	type ITestFluidObject,
+	waitForContainerConnection,
+	timeoutAwait,
+} from "@fluidframework/test-utils/internal";
+
+// eslint-disable-next-line import/no-internal-modules
+import { loadOffline, getPendingOps } from "./offline/offlineTestsUtils.js";
+
+const waitForSummary = async (
```

Comment:
Consider renaming this to something like `loadSummarizerAndSummarize` to indicate that it creates a summarizer as well. It wasn't clear when I was looking at the call sites and I got a little confused.